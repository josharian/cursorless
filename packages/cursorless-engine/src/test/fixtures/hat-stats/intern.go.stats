nTokens: 228
nHats:
	mean: 100%
	std: 0%
	min: 100%
	max: 100%
	spark: 0              25               50               75             █ 100

nPenalty0:
	mean: 15%
	std: 0%
	min: 15%
	max: 15%
	spark: 0        █     25               50               75               100

nPenalty1:
	mean: 85%
	std: 0%
	min: 85%
	max: 85%
	spark: 0              25               50               75     █         100

nPenalty2:
	mean: 0%
	std: 0%
	min: 0%
	max: 0%
	spark: 0 █            25               50               75               100

