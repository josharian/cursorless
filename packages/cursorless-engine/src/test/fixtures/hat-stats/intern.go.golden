*   *       *           *      * *
// Package intern interns strings.
[] [-----] [----] [-----] [-----]#

x                               x
*     *      *     *  *       * *
// Interning is best effort only.
[] [-------] [] [--] [----] [--]#

v
*         *   *       * *      *    *
// Interned strings may be removed automatically
[] [------] [-----] [-] [] [-----] [-----------]

w   x  x                           v
*   *  *    *     *          *     *
// at any time without notification.
[] [] [-] [--] [-----] [----------]#

o   x              x  x      x      x
*   *        *     *  *      *      *
// All functions may be called concurrently
[] [-] [-------] [-] [] [----] [----------]

#              x     v   x      vw
*  *           *     *   *      **
// with themselves and each other.
[] [--] [--------] [-] [--] [---]#

         v
   *     *
package intern
[-----] [----]

         v  x
  *    * *  *
import "sync"
[----] #[--]#

 x     v  w  o x      o  # v
 *     *  *  * *   *  *  * *  *
var pool sync.Pool = sync.Pool{
[-] [--] [--]#[--] # [--]#[--]#

   x   x       v      x  v
   **  *  **   *      ** *
␉New: func() interface{} {
 [-]# [--]## [-------]## #

     v     x x  x      x  w    x
     *     * *  **     ** *    *
␉	return make(map[string]string)
  [----] [--]#[-]#[----]#[----]#

 x
 **
␉},
 ##

v
*
}
#

^   o        w    xx        w^
*   *        *    **        **
// String returns s, interned.
[] [----] [-----] ## [------]#

   v      vvv  #    v  ^     w
   *      ***  *    *  *     *
func String(s string) string {
[--] [----]## [----]# [----] #

 x xx  w  0 v ww>o v x  w   x 0    o
 * **  *  * * **** * *  *   * *    *
␉m := pool.Get().(map[string]string)
 # ## [--]#[-]####[-]#[----]#[----]#

 wv  v vv vvwv
 **  * ** ****
␉c, ok := m[s]
 ## [] ## ####

  x  w o
  *  * *
␉if ok {
 [] [] #

     w~ o #w#
     ** * ***
␉	pool.Put(m)
  [--]#[-]###

   w     o
   *     *
␉	return c
  [----] #

 w
 *
␉}
 #

 owow w #
 **** * *
␉m[s] = s
 #### # #

  o  + # ^#^
  *  * * ***
␉pool.Put(m)
 [--]#[-]###

  o     ^
  *     *
␉return s
 [----] #

o
*
}
#

0   #       ^    x    v       # w      ww    #
*   *       *    *    *       * *      **    *    b
// Bytes returns b converted to a string, interned.
[] [---] [-----] # [-------] [] # [----]# [------]#

 0    ^   0v oo 0  0      o #
 *    *   ** ** *  *      * *
func Bytes(b []byte) string {
[--] [---]## ##[--]# [----] #

 ^ wo    o  ^ >> ~ o #  o   # >    ~
 * **    *g * **r* * *  *   * *    *
␉m := pool.Get().(map[string]string)
 # ## [--]#[-]####[-]#[----]#[----]#

 #o  o o# 0^ ~    +w+^
 **  * ** ** *    ****
␉c, ok := m[string(b)]
 ## [] ## ##[----]####

  v  # ^
  *  * *
␉if ok {
 [] [] #

     #  >  >
     *p * b*b
␉	pool.Put(m)
  [--]#[-]###

   0     ^
   *     *
␉	return c
  [----] #

 #
 *
␉}
 #

 0 #^      # o
 * **      *g*g
␉s := string(b)
 # ## [----]###

 ~0>0 0 ~
 **** * *
␉m[s] = s
 #### # #

  ^    ~  +
  *  y * r*r
␉pool.Put(m)
 [--]#[-]###

  >     +
  *     *
␉return s
 [----] #

^
*
}
#

